topics/go/algorithms/data/hash/hash.go

```diff
@@ -0,0 +1,107 @@
+// Package hash implements a hash table
+package hash
+
+import (
+	"fmt"
+	"hash/maphash"
+)
+
+const (
+	numBuckets = 256 // Number of buckets in hash
+)
+
+// entry in the hash table
+type entry struct {
+	key   string
+	value int
+}
+
+// Hash is a simple Hash table
+type Hash struct {
+	buckets [][]*entry
+	h       maphash.Hash
```

Comment:
This variable `h` doesn't work for readability.  Later on you have code that looks like `h.h.Reset()`

I was hope we could use embedding but that won't work. :(

I think you call it `hash`. then you will have `h.hash.Reset()`

Reply:
OK